src/webui/abstractwebapplication.cpp

```diff
@@ -307,10 +308,12 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (Utils::Net::isLANAddress(env_.clientAddress))
```

Comment:
I'm not sold on the benefits of doing so. Reserved local ips aren't routable over the internet anyway, so adding that additional check would break the feature for certain client environments.